### DISCUSSION TITLE
OMVIF Test Tool Device Discovery

### DISCUSSION THREAD

**Discussion by harlyn-porter** [onvif/testspecs#27] [Q&A] [answered]
How do we configure the Test tool to scan specific IP address range(s) to locate devices?  We have several subnets on our networks, and the devices are located on specific subnets.  We are able to locate devices when we probe specific IP addresses, but the Discovery tool is unable to locate any devices when we scan the available NICs.

**Comment by baldvin137** [comment, marked as answer]
The recommended setup is a dedicated network, so I don't have practical experience with using subnets. But I have some theoretical suggestions. 

ONVIF discovery is based on WS-Discovery where the default TTL is 1. Probing is different as it goes to a specific IP address. I'm not sure what the TTL value is for the DTT, but I would not expect that it is configurable in the current version. But if it is 1 and not configurable, then it might be an improvement suggestion for the DTT. I don't know if it is a good idea, but the DTT work group could take a look at the suggestion. However, there is a reason why a single dedicated net is recommended the recommended setup. E.g. multicast tests and UDP related tests can become unreliable if not testing on a dedicated network.

**Reply by harlyn-porter** [reply]
Unfortunately we do not have the option to have a dedicated network. The discovery tool resides on a server in the cloud, and the devices are all connected via VPN tunnels. Al of the network communications are handled by our firewall, so we cannot select the network through the discovery tool.
